app/actions/handlers.py

```diff
@@ -14,85 +15,58 @@
 state_manager = IntegrationStateManager()
 
 
-async def filter_and_transform(vehicles, transmissions, integration_id, action_id):
+def extract_gmt_offsets(transmissions):
+    result = {
+        "transmissions": reduce(
+            lambda acc, item: acc.setdefault(item.collar_serial_num, item.gmt_offset) or acc,
+            transmissions,
+            {}
+        )
+    }
+
+    return result["transmissions"]
```

Comment:
I don't think that reduce call is going to succeed. You might want to write a unit test to see what will happen. But to simplify this you could use this instead.

```suggestion
    accumulator = {}
    for item in transmissions:
        accumulator.setdefault(item.collar_serial_num, item.gmt_offset)
    return accumulator
```